server/apps/immich/src/api-v1/asset/dto/create-asset.dto.ts

```diff
@@ -11,7 +11,8 @@ export class CreateAssetDto {
   deviceId!: string;
 
   @IsNotEmpty()
-  @ApiProperty({ enumName: 'AssetTypeEnum', enum: AssetType })
+  @IsEnum(AssetType)
+  @ApiProperty({ enum: Object.keys(AssetType) })
```

Comment:
Why does this need to be changed? If anything it would need to be `Object.values(AssetType)`, but I believe it was working fine before with `{ enumName, enum }`

Reply:
Object.keys() and Object.values() happen to have the same result in this instance, but you're right it should be Object.values().

When I used `{ enumName, enum }` as before it would generate a malfunctioning API for Typescript as shown below.

```js
// With { enumName, enum }
localVarFormParams.append('assetType', new Blob([JSON.stringify(assetType)], { type: "application/json", }));

// With enum: Object.keys(AssetType)
localVarFormParams.append('assetType', assetType as any);

```

Reply:
Is this actually malfunctioning?

Reply:
Well malfunctioning is probably the wrong word. It does serialize, but you would have to deserialize it as JSON on the server but only for requests from the web. Now that I've changed the generation of the SDK with the latest commits, this is no longer an issue.

Reply:
The web doesn't even use this in the codebase right now AFAIK.

Reply:
It's marked as `TODO: should probably use the @api SDK` [here](https://github.com/immich-app/immich/blob/74cd3d66c62aaed7d6d324532b727706116d18d3/web/src/lib/utils/file-uploader.ts#L71). Once these changes are made, we should use the generated API.